system/lib/wasmfs/wasmfs.cpp

```diff
@@ -56,16 +56,52 @@ WasmFS::WasmFS() : rootDirectory(initRootDirectory()), cwd(rootDirectory) {
 // (Note that this means we can't find leaks inside WasmFS code itself, but that
 // seems fundamentally impossible for the above reasons, unless we made LSan log
 // its findings in a way that does not depend on normal file I/O.)
-__attribute__((weak)) extern "C" void __lsan_do_leak_check(void) {
-}
+__attribute__((weak)) extern "C" void __lsan_do_leak_check(void) {}
 
 extern "C" void wasmfs_flush(void) {
   // Flush musl libc streams.
   fflush(0);
 
-  // Flush our own streams. TODO: flush all backends.
+  // Flush our own streams.
   (void)SpecialFiles::getStdout()->locked().flush();
   (void)SpecialFiles::getStderr()->locked().flush();
+
+  std::vector<std::shared_ptr<Directory>> toFlush;
+  toFlush.push_back(wasmFS.getRootDirectory());
+
+  while (!toFlush.empty()) {
+    auto dir = toFlush.back();
+    toFlush.pop_back();
+
+    auto lockedDir = dir->locked();
+    Directory::MaybeEntries entries = lockedDir.getEntries();
+#ifndef NDEBUG
+    if (int err = entries.getError()) {
+      std::string errorMessage =
+        "Non-fatal error code " + std::to_string(err) +
+        " while flushing directory: " + lockedDir.getName(dir);
+      emscripten_console_error(errorMessage.c_str());
+      continue;
+    }
+#endif
+    for (auto entry : *lockedDir.getEntries()) {
```

Comment:
```suggestion
    for (auto entry : *entries) {
```
This was saved to a local already.